cmd/notation/policy/import.go

```diff
@@ -0,0 +1,96 @@
+package policy
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/notaryproject/notation-go/dir"
+	"github.com/notaryproject/notation-go/log"
+	"github.com/notaryproject/notation-go/verifier/trustpolicy"
+	"github.com/notaryproject/notation/cmd/notation/internal/cmdutil"
+	"github.com/notaryproject/notation/internal/cmd"
+	"github.com/notaryproject/notation/internal/osutil"
+	"github.com/spf13/cobra"
+)
+
+type importOpts struct {
+	cmd.LoggingFlagOpts
+	filePath string
+	force    bool
+}
+
+func importCmd() *cobra.Command {
+	var opts importOpts
+	command := &cobra.Command{
+		Use:   "import [flags] <file_path>",
+		Short: "[Preview] Import trust policy configuration from a JSON file",
+		Long: `[Preview] Import trust policy configuration from a JSON file.
+
+** This command is in preview and under development. **
+
+Example - Import trust policy configuration from a file:
+  notation policy import my_policy.json
+`,
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.filePath = args[0]
+			return runImport(cmd, opts)
+		},
+	}
+	opts.LoggingFlagOpts.ApplyFlags(command.Flags())
+	command.Flags().BoolVar(&opts.force, "force", false, "override the existing trust policy configuration, never prompt")
+	return command
+}
+
+func runImport(command *cobra.Command, opts importOpts) error {
+	ctx := opts.LoggingFlagOpts.SetLoggerLevel(command.Context())
+	logger := log.GetLogger(ctx)
+	// optional confirmation
+	var mode os.FileMode = 0644
+	policyPath := filepath.Join(dir.UserConfigDir, dir.PathTrustPolicy)
+	if fi, err := os.Stat(policyPath); err == nil {
+		mode = fi.Mode()
+		confirmed, err := cmdutil.AskForConfirmation(os.Stdin, "Existing trust policy configuration found, do you want to overwrite it?", opts.force)
+		if err != nil {
+			return err
+		}
+		if !confirmed {
+			return nil
+		}
+	}
+
+	// read configuration
+	policyJson, err := os.ReadFile(opts.filePath)
+	if err != nil {
+		return fmt.Errorf("failed to read trust policy configuration: %w", err)
+	}
+
+	// parse and validate
+	var doc trustpolicy.Document
+	if err = json.Unmarshal(policyJson, &doc); err != nil {
+		return fmt.Errorf("failed to parse trust policy configuration: %w", err)
+	}
+
+	if err := doc.Validate(); err != nil {
+		for _, p := range doc.TrustPolicies {
+			tmp := doc
+			tmp.TrustPolicies = []trustpolicy.TrustPolicy{p}
+			if err = tmp.Validate(); err != nil {
+				logger.Errorf("failed to validate trust policy %q:\n", p.Name)
+				if content, err := json.Marshal(&p); err == nil {
+					logger.Errorf("\n%s\n", string(content))
+				}
+				return err
+			}
+		}
```

Comment:
We could just return the err directly. The error messages of notation-go `Validate()` https://github.com/notaryproject/notation-go/blob/fb79e6df1aa8cacc8c9f4e917e36ec6dc0403f37/verifier/trustpolicy/trustpolicy.go#L157 ought to provide the details of the failure. If not, please file an issue in notation-go, and we need to improve those error messages there.
```suggestion
		return err
```

Reply:
implemented as suggested